core/services/ocr2/plugins/ocr2vrf/fixed_blockhash_provider.go

```diff
@@ -0,0 +1,109 @@
+package ocr2vrf
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/smartcontractkit/ocr2vrf/types"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm/client"
+)
+
+// Implementation of Blockhashes interface that returns blockhashes with fixed-sized windows relative to current block height
+type FixedBlockhashProvider struct {
```

Comment:
Minor: this type does not need to be exported if you have a constructor (i.e a `New*` method). We can keep this private and the `New*()` method will return a `types.Blockhashes` rather than this concrete type.